### PR TITLE
Refactor `AsChangeset!` now that macros in type positions are stable

### PR DESCRIPTION
**Pull request by sgrif**
Cleaner code, less recursion. We can just return the type now!

### CODE REVIEW

**Comment by sgrif** [comment]
Review? @diesel-rs/contributors

File: diesel_tests/Cargo.toml [hunk 1]
```diff
@@ -31,6 +31,10 @@ stable_sqlite = ["with-syntex", "sqlite", "diesel_codegen_syntex/sqlite"]
 unstable_postgres = ["unstable", "postgres", "diesel_codegen/postgres"]
 unstable_sqlite = ["unstable", "sqlite", "diesel_codegen/sqlite"]
 
+[lib]
+name = "integration_tests"
+path = "tests/lib.rs"
+
```
Comment (killercup) [inline diesel_tests/Cargo.toml]: Huh? Why is this now necessary when it wasn't before?

Comment (sgrif) [inline diesel_tests/Cargo.toml]: Oh whoops that wasn't supposed to get committed. I do that when I need to do `-Ztrace-macros`

**Comment by sgrif** [comment]
I don't have a list, no. I usually add a FIXME comment if something is significantly worse off without a feature.

**Pull request merged**